### PR TITLE
added force flag and update upgrade_metadata json file 

### PR DESCRIPTION
**Pull request by shaik80**
Signed-off-by: shaik80 <smudassi@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
In this PR force flag is added to migrate pg and the upgrade_metadata.json file is updated if pg migrated.

### :chains: Related Resources
https://chefio.atlassian.net/browse/SHIELD-16?atlOrigin=eyJpIjoiODUwMmQ1YjYwZmY3NGVmZjk4OWIzOTJlYzIyYzRhZTQiLCJwIjoiaiJ9

### :+1: Definition of Done
1. It should update JSON file if  pg migrate cmd is executed 
2. It should check pg migrate cmd is executed or not, if it is executed prompt message to force execute (Y/N)
3. To avoid prompt message -f flag can be used

### :athletic_shoe: How to Build and Test the Change
```
automate ✗   hab studio enter
[20][default:/src:0]# start_all_services 
[20][default:/src:0]# rebuild components/automate-deployment/
[20][default:/src:0]# rebuild components/automate-cli
[20][default:/src:0]# chef-automate upgrade run --major
[20][default:/src:0]# chef-automate upgrade status
[20][default:/src:0]# chef-automate post-major-upgrade migrate --data=PG -f
or 
[20][default:/src:0]# chef-automate post-major-upgrade migrate --data=PG
migrate_pg is already executed,do you want to force execute.
Press y to agree, n to disagree? [y/n]: y
```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Review by vinay033 — APPROVED** [review]

**Review by vivekshankar1 — COMMENTED** [review]
Declare all strings as const

**Review by shivangi-singh1 — APPROVED** [review]

File: components/automate-cli/cmd/chef-automate/migration_pg.go [hunk 1]
```diff
@@ -60,30 +66,32 @@ var migrateCmd = &cobra.Command{
...
-	migratePgCmd.PersistentFlags().StringVar(&migrateDataCmdFlags.data, "data", "", "data")
-	migratePgCmd.PersistentFlags().BoolVarP(&migrateDataCmdFlags.autoAccept, "", "y", false, "auto-accept")
-	return migratePgCmd
+		RunE:  runMigrateDataCmd,
+	}
+	migrateDataCmd.PersistentFlags().BoolVar(&migrateDataCmdFlags.check, "check", false, "check")
+	migrateDataCmd.PersistentFlags().StringVar(&migrateDataCmdFlags.data, "data", "", "data")
+	migrateDataCmd.PersistentFlags().BoolVarP(&migrateDataCmdFlags.autoAccept, "", "y", false, "auto-accept")
+	migrateDataCmd.PersistentFlags().BoolVarP(&migrateDataCmdFlags.skipStorageCheck, "skip-storage-check", "s", false, "skip storage check")
+	migrateDataCmd.Flags().BoolVarP(&migrateDataCmdFlags.forceExecute, "force", "f", false, "fore-execute")
```
Comment (vivekshankar1) [inline components/automate-cli/cmd/chef-automate/migration_pg.go:93]:
```suggestion
	migrateDataCmd.Flags().BoolVarP(&migrateDataCmdFlags.forceExecute, "force", "f", false, "force-execute")
```

File: components/automate-cli/cmd/chef-automate/migration_pg.go [hunk 2]
```diff
@@ -100,11 +108,38 @@ func runCleanup(cmd *cobra.Command, args []string) error {
...
+					)
+					if err != nil {
+						return err
+					} else {
+						isExecuted = false
+					}
+				}
+			}
+
+			if !isExecuted {
```
Comment (vivekshankar1) [inline components/automate-cli/cmd/chef-automate/migration_pg.go:136]: change this to if else block

File: components/automate-cli/cmd/chef-automate/migration_pg.go [hunk 3]
```diff
@@ -115,51 +150,83 @@ func runCleanup(cmd *cobra.Command, args []string) error {
...
-			if err != nil {
-				return err
-			}
+		ci, err := majorupgradechecklist.NewPostChecklistManager(AUTOMATE_VERSION)
+		if err != nil {
+			return err
 		}
 
-		oldPgVersion, err := pgVersion(OLD_PG_DATA_DIR + "/PG_VERSION")
+		isExecuted, err := ci.ReadPostChecklistById("migrate_pg", majorupgradechecklist.UPGRADE_METADATA)
```
Comment (vivekshankar1) [inline components/automate-cli/cmd/chef-automate/migration_pg.go:167]: replace string by const

File: components/automate-cli/cmd/chef-automate/migration_pg.go [hunk 4]
```diff
@@ -115,51 +150,83 @@ func runCleanup(cmd *cobra.Command, args []string) error {
...
+				} else {
+					isExecuted = false
+				}
+			}
+		}
+
+		if !isExecuted {
+			if !migrateDataCmdFlags.check && !migrateDataCmdFlags.autoAccept {
+				err := promptCheckList(
+					"it will start the migration immediately after check.\nPress y to agree, n to disagree? [y/n]",
```
Comment (vivekshankar1) [inline components/automate-cli/cmd/chef-automate/migration_pg.go:190]: start sentence with caps

**Review by vivekshankar1 — COMMENTED** [review]
declare all strings as const

**Review by vivekshankar1 — APPROVED** [review]
looks good, thanks @shaik80

**Pull request merged**